platform_channels/android/app/src/main/kotlin/dev/flutter/platform_channels/MainActivity.kt

```diff
@@ -50,8 +48,31 @@ class MainActivity : FlutterActivity() {
                 .setMessageHandler { message, reply ->
                     if (message == "getImage") {
                         val inputStream: InputStream = assets.open("eat_new_orleans.jpg")
-                        reply.reply(inputStream.readBytes());
+                        reply.reply(inputStream.readBytes())
                     }
                 }
+
+        // Registers a MessageHandler for BasicMessageChannel using StringCodec to receive a message
+        // from Dart and send a reply.
+        BasicMessageChannel(flutterEngine.dartExecutor, "stringCodecDemo", StringCodec.INSTANCE)
+                .setMessageHandler { message, reply ->
+                    reply.reply("$message from Android")
+                }
+
+        // Registers a MessageHandler for BasicMessageChannel using JSONMessageCodec to receive a message
+        // from Dart and send a reply.
+        BasicMessageChannel(flutterEngine.dartExecutor, "jsonMessageCodecDemo", JSONMessageCodec.INSTANCE)
+                .setMessageHandler { message, reply ->
+                    reply.reply("${message.toString()} from Android")
+                }
```

Comment:
While I love that you've kept things simple throughout this sample, I wonder if there's a missed opportunity here to show JSON codec sending more structured data. Would it be possible to show a message containing more than one value being sent back and forth here?

Reply:
Sure, I'll look to modify example a bit or come with something new where we can use `Map` to send the data across the platform.

Reply:
Hey @RedBrogdon, I was thinking to do a build a simple Dogs and Cats list sample, where you can add the type of Animal whether it's cat or dog and their breed, and delete any item from the list. The list will be stored on the native side.
Every time user adds a new item to the list, we can send a `Map` to the native side and add it to the list. Once, the item is added to the list, we can simply send the list using the channel for which we will have a `MessageHandler` at the dart side. For deleting the item, we can have another `BasicMessageChannel` with `BinaryCodec` where we can convert the index of `List` -> `String` -> `ByteData` to delete any item from the list. We can remove the usage of `StringCodec` since it's pretty straight forward. WDYT?

Reply:
I think that's a great plan. It would also allow you to show calls that *originate* on the native side, which you did with an EventChannel but (I believe) haven't yet done with a MethodChannel.

Reply:
Also, thanks again for understanding that last week was a busy week for us. Thursday and Friday will be holidays for Googlers in the United States, but I'm back at normal speed. 😄

Reply:
Awesome, I'll do all the required changes in the next commit.